lib/puppet/gettext/config.rb

```diff
@@ -35,7 +46,31 @@ def self.translation_mode(conf_path)
   # @param file_format [Symbol] translation file format to use, either :po or :mo
   # @return true if initialization succeeded, false otherwise
   def self.initialize(conf_file_location, file_format)
-    # Bypass gettext until we can resolve a performance regression related to it, PUP-8009.
-    return false
+    unless file_format == :po || file_format == :mo
+      raise Puppet::Error, "Unsupported translation file format #{file_format}; please use :po or :mo"
+    end
+
+    return false unless @found
+
+    if conf_file_location && File.exists?(conf_file_location)
```

Comment:
should we also cache whether the file exists?

Reply:
That is handled automatically as part of environment caching. If we use environment caching, then the Module will only be created/initialized when the cache is invalid.

Reply:
Oh I see. Ok.